ilp-core/src/main/java/org/interledger/core/InterledgerAddress.java

```diff
@@ -140,6 +140,18 @@ default boolean startsWith(final InterledgerAddress interledgerAddress) {
     return this.getValue().startsWith(interledgerAddress.getValue());
   }
 
+  /**
+   * <p>Tests if this InterledgerAddress starts with the specified {@code addressPrefix}.</p>
```

Comment:
I think it would be useful to clarify in the Javadoc here the somewhat surprising case of an ILP address `g.foo.bob` and a prefix of `g.foo.bob`, which will evalue `false` here.

I'm actually still on the fence if this is the correct behavior. I _think_ it is, but it's surprising nonetheless. For example, for a minute I thought this was the right implementation of this method:

```java
return this.getValue().startsWith(addressPrefix.getValue());
```

But after thinking about it a bit more, I think the way you did it below is technically correct due to calling `getPrefix`, but it's still surprising which is giving me pause.

Reply:
@adrianhopebailie Do you have any opinion as to what the correct behavior should be here?

Reply:
@sudheesh001 after thinking a bit more about this last night, I realized I made an error in my comment above. 

It occurred to me that proposed implementation in this PR of [startsWith](https://github.com/hyperledger/quilt/pull/211/files#diff-e147a50f73f571a8f86bff72f2299dadR150) is incorrect. My rationale is that this method is being called on an Interledger Address, with an input of a `prefix`. In other words, the method is allowing developers to ask, "Does this ILP address start with this prefix?" 

I think the proper answer to this is _not_ to compare the supplied input against the ILP Address's prefix (which is instead a derived property of the ILP address). Instead, the question above is more properly answered by computing whether or not the entire address starts with the prefix. In other words, we should be comparing the input against the entire address itself.

So, I think the proper way to implement this method is actually like this:

```java
return this.getValue().startsWith(addressPrefix.getValue());
``` 

Does that reasoning make sense?

This also should resolve any "surprise" encountered by certain unit tests. For example: `g.foo.com` _does_ technically start with the prefix `g.foo.com`.

Reply:
Ah I see. Thinking about it a little more I think this behaviour is consistent with what someone using `startsWith()` would expect. Making the changes and sending in a PR.